templates/src/main/java/org/teavm/flavour/templates/parsing/Parser.java

```diff
@@ -114,6 +115,47 @@ public boolean wasSuccessful() {
 
     public List<TemplateNode> parse(Reader reader, String className) throws IOException {
         source = new Source(reader);
+        source.setLogger(new Logger() {
```

Comment:
I don't think it's the right way. There's TeaVM diagnostics which can be used to report errors and warnings. Also, I doubt if it's really necessary to report debug and info messages.

Reply:
OK, I will try adding filenames to the diagnostic.  I'll update the PR when it is ready.

Reply:
I couldn't figure out how to extract a list of errors or warnings from Jericho.  I did figure out a way to preserve the original logger log level so it won't report debug and info messages.  Updated PR coming shortly for your review.